include/prometheus/exposer.h

```diff
@@ -25,7 +26,12 @@ class Exposer {
 
  private:
   std::unique_ptr<CivetServer> server_;
-  std::vector<std::weak_ptr<Collectable>> collectables_;
+  // \todo maybe better create here class with mutex?
+  //       maybe best solution will be lock free list
+  //       But in any case container need syncronization defence
+  //       One thread can do RegisterCollectable and another thread in server can do collect from container
+  std::mutex collectables_guard_;
```

Comment:
A better way to solve this would be to have exposer take a container of collectables in its constructor. This would alleviate the need for locks. The list of collectables should be finite anyway, I don't think we need the dynamics of the `RegisterCollectable()` function.

Reply:
Okay. You are right. I will do fix.